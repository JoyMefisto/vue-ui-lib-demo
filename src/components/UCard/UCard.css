.ui-card {
  border: 1px solid var(--color-card-border);
  border-radius: var(--radius-md);
  background: var(--color-card-bg);
  overflow: hidden;
}

.ui-card__header,
.ui-card__footer {
  padding: var(--spacing-sm) var(--spacing-md);
  font-weight: 600;
}

.ui-card__content {
  padding: var(--spacing-md);
}
