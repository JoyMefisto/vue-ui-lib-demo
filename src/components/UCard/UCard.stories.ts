import type { Meta, StoryObj } from '@storybook/vue3'
import UCard from './UCard.vue'

const meta: Meta<typeof UCard> = {
  title: 'UI/Card',
  component: UCard,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof UCard>

// 1. Basic card with only content
export const Basic: Story = {
  render: () => ({
    components: { UCard },
    template: `
      <UCard>
        This is a simple card with only content.
      </UCard>
    `,
  }),
}

// 2. Card with header + content
export const WithHeader: Story = {
  render: () => ({
    components: { UCard },
    template: `
      <UCard>
        <template #header>Card Header</template>
        Some content inside the card.
      </UCard>
    `,
  }),
}

// 3. Card with header + footer
export const WithHeaderAndFooter: Story = {
  render: () => ({
    components: { UCard },
    template: `
      <UCard>
        <template #header>Title</template>
        Card body content goes here.
        <template #footer>Footer info</template>
      </UCard>
    `,
  }),
}

// 4. Long content card
export const LongContent: Story = {
  render: () => ({
    components: { UCard },
    template: `
      <UCard>
        <template #header>Article</template>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
        <p>Aliquam ut feugiat lorem. Suspendisse potenti.</p>
        <template #footer>Read more...</template>
      </UCard>
    `,
  }),
}

// 5. Nested content
export const Nested: Story = {
  render: () => ({
    components: { UCard },
    template: `
      <UCard>
        <template #header>Dashboard Widget</template>
        <UCard>
          <template #header>Nested Card</template>
          Nested content here.
        </UCard>
        <template #footer>Updated 1h ago</template>
      </UCard>
    `,
  }),
}
