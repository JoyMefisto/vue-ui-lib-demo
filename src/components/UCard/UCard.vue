<script setup lang="ts">
import '@/styles/theme.css'
import '@/styles/base.css'
import './UCard.css'
</script>
<template>
  <div class="ui-card">
    <header class="ui-card__header">
      <slot name="header" />
    </header>
    <div class="ui-card__content">
      <slot />
    </div>
    <footer class="ui-card__footer">
      <slot name="footer" />
    </footer>
  </div>
</template>
