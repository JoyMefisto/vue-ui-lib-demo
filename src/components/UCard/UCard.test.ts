import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import UCard from './UCard.vue'

describe('UCard', () => {
  it('renders default slot content', () => {
    const wrapper = mount(UCard, {
      slots: {
        default: 'Main content',
      },
    })

    expect(wrapper.find('.ui-card__content').text()).toContain('Main content')
  })

  it('renders header slot when provided', () => {
    const wrapper = mount(UCard, {
      slots: {
        header: 'Card Header',
      },
    })

    expect(wrapper.find('.ui-card__header').text()).toBe('Card Header')
  })

  it('renders footer slot when provided', () => {
    const wrapper = mount(UCard, {
      slots: {
        footer: 'Footer text',
      },
    })

    expect(wrapper.find('.ui-card__footer').text()).toBe('Footer text')
  })

  it('renders all slots together', () => {
    const wrapper = mount(UCard, {
      slots: {
        header: 'Header',
        default: 'Body content',
        footer: 'Footer',
      },
    })

    expect(wrapper.find('.ui-card__header').text()).toBe('Header')
    expect(wrapper.find('.ui-card__content').text()).toBe('Body content')
    expect(wrapper.find('.ui-card__footer').text()).toBe('Footer')
  })

  it('renders empty header/footer when not provided', () => {
    const wrapper = mount(UCard, {
      slots: {
        default: 'Only content',
      },
    })

    expect(wrapper.find('.ui-card__header').text()).toBe('')
    expect(wrapper.find('.ui-card__footer').text()).toBe('')
  })

  it('matches snapshot', () => {
    const wrapper = mount(UCard, {
      slots: {
        header: 'Header',
        default: 'Content',
        footer: 'Footer',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
