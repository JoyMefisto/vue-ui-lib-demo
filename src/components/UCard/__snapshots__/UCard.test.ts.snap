// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UCard > matches snapshot 1`] = `
"<div class="ui-card">
  <header class="ui-card__header">Header</header>
  <div class="ui-card__content">Content</div>
  <footer class="ui-card__footer">Footer</footer>
</div>"
`;
